tests/unit/models/test_evc_deploy.py

```diff
@@ -1345,6 +1345,97 @@ def test_run_bulk_sdntraces(self, put_mock):
         self.assertEqual(result, {"result": []})
 
+    def test_map_dl_vlan(self):
+        """Test map_dl_vlan"""
+        cases = {0: None, 'untagged': None, 'any': 1, '4096/4096': 1, 10: 10}
+        for value, mapped in cases.items():
+            result = EVCDeploy.map_dl_vlan(value)
+            assert result == mapped
+
+    @patch("requests.put")
+    def test_run_bulk_sdntraces_special_vlan(self, put_mock):
+        """Test run_bulk_sdntraces method for bulk request."""
+        evc = self.create_evc_inter_switch()
+        response = MagicMock()
+        response.status_code = 200
+        put_mock.return_value = response
+
+        expected_endpoint = f"{SDN_TRACE_CP_URL}/traces"
+        expected_payload = [
+                            {
+                                'trace': {
+                                    'switch': {'dpid': 1, 'in_port': 2}
+                                }
+                            }
+                        ]
+
+        evc.uni_a.user_tag.value = 'untagged'
+        EVCDeploy.run_bulk_sdntraces([evc.uni_a])
+        put_mock.assert_called_with(
+                                    expected_endpoint,
+                                    json=expected_payload,
+                                    timeout=30
+                                )
+        args = put_mock.call_args[1]['json'][0]
+        assert 'eth' not in args
+
+        evc.uni_a.user_tag.value = 0
+        EVCDeploy.run_bulk_sdntraces([evc.uni_a])
+        put_mock.assert_called_with(
+                                    expected_endpoint,
+                                    json=expected_payload,
+                                    timeout=30
+                                )
+        args = put_mock.call_args[1]['json'][0]['trace']
+        assert 'eth' not in args
+
+        expected_payload[0]['trace']['eth'] = {'dl_type': 0x8100, 'dl_vlan': 1}
+        evc.uni_a.user_tag.value = 'any'
+        EVCDeploy.run_bulk_sdntraces([evc.uni_a])
+        put_mock.assert_called_with(
+                                    expected_endpoint,
+                                    json=expected_payload,
+                                    timeout=30
+                                )
+        args = put_mock.call_args[1]['json'][0]['trace']
+        assert args['eth'] == {'dl_type': 33024, 'dl_vlan': 1}
+
+        evc.uni_a.user_tag.value = '4096/4096'
+        EVCDeploy.run_bulk_sdntraces([evc.uni_a])
+        put_mock.assert_called_with(
+                                    expected_endpoint,
+                                    json=expected_payload,
+                                    timeout=30
+                                )
+        args = put_mock.call_args[1]['json'][0]['trace']
+        assert args['eth'] == {'dl_type': 33024, 'dl_vlan': 1}
+
+        expected_payload[0]['trace']['eth'] = {
+            'dl_type': 0x8100,
+            'dl_vlan': 10
+            }
+        evc.uni_a.user_tag.value = '10/10'
```

Comment:
It's good to have this covered.

It'd be great if you could cover a less trivial mask example too, like "5/2" -> 0 and "5/3" -> 1.

Reply:
@viniarck, now I have a doubt. In the case "5/2 -> 0 and the similar, i.e. "20/10" --> "n1/n2" with [n1 & (n2 $4095)] == 0, the mapped value should be 0? We agreed that ` dl_vlan` would be in range [1,4095].

Reply:
Right, @gretelliz, `"5/2"` will result in 0, but since you have this if `if uni_dl_vlan`, then it'll be sent without a vlan header. This case here is a corner case, since we don't have validation for the mask length in the API yet and we're allowing any of the 12 bits to be 0s or 1s, then it can result in 0 (one day we might restrict that further too, for now we're keeping it very open since vlan range discussion is still going on). and `"5/3"` -> 1, which is why it's good to cover these cases with unit cases, since in the future if we decide to further restrict more then unit tests will catch it.

Reply:
Great, everything is clear. Thank you.